src/components/editor/mirror-mark/index.js

```diff
@@ -126,29 +130,51 @@ class MirrorMark {
         }
       },
       preview: function() {
-        const setPreviewMode = function(cm) {
+        const setPreviewMode = function(_this) {
           var converter = new Markdown.Converter()
-          var wrap = cm.getWrapperElement()
+          var wrap = _this.cm.getWrapperElement()
           wrap.className += ' CodeMirror-has-preview'
-          var previewNodes = wrap.getElementsByClassName("CodeMirror-preview")
+          var previewNodes = wrap.getElementsByClassName('CodeMirror-preview')
           var previewNode
 
           if(previewNodes.length == 0) {
             var previewNode = document.createElement('div')
-            previewNode.className = "CodeMirror-preview"
+            previewNode.className = 'CodeMirror-preview'
             wrap.appendChild(previewNode)
           } else {
             previewNode = previewNodes[0]
           }
 
-          previewNode.innerHTML = converter.makeHtml(cm.getValue())
+          previewNode.innerHTML = converter.makeHtml(_this.cm.getValue())
+          var anchors = document.getElementsByTagName('a')
+          for(var i = 0; i < anchors.length; i++){
```

Comment:
this click handling can be removed from here now right?

Reply:
This is needed to interrupt redirection (so the app doesn't refresh), and to change the icon.

Reply:
Yeah, preventDefault should take care of it. I didn't need to touch mirror mark at all to get the poc code working for this stuff.

Reply:
Done.